playground.py

```diff
@@ -0,0 +1,161 @@
+
+import torch
+
+import torch.distributions
+
+from typing import Dict
+
+
+class Transform:
+    def __init__(self):
+        print(f"{self.__class__.__name__} __init__")
+
+    def __call__(self, *args, **kwargs):
+        print(f"{self.__class__.__name__} __call__")
+
+
+from typing import Callable, Union
+
+class Parameter:
+    def __init__(self, count: Union[int, Callable[[Transform], int]]):
+        self.count = count
+
+    def initialize(self, transform: Transform):
+        if isinstance(self.count, Callable):
+            self.count = self.count(transform)
+
+    def constrain(self, unconstrained: torch.Tensor) -> torch.Tensor:
+        raise NotImplementedError
+
+
+class Real(Parameter):
+    def constrain(self, unconstrained: torch.Tensor) -> torch.Tensor:
+        return unconstrained
+
+
+class Positive(Parameter):
+    def constrain(self, unconstrained: torch.Tensor) -> torch.Tensor:
+        return torch.exp(unconstrained)
```

Comment:
I think we should have something like `SoftPositive` that saturates at a certain value.